tracee-rules/engine/engine.go

```diff
@@ -107,17 +133,102 @@ func (engine *Engine) consumeSources(done <-chan bool) {
 						}
 					}
 				}
+				engine.signaturesMutex.RUnlock()
 				engine.inputs.Tracee = nil
 			} else if event != nil {
-				for _, s := range engine.signaturesIndex[types.SignatureEventSelector{Source: "tracee", Name: event.(tracee.Event).EventName}] {
+				engine.signaturesMutex.RLock()
+				assertedEvt, ok := event.(tracee.Event)
```

Comment:
for clarity, could rename `assertedEvt` to `traceeEvt` (or `traceeEvent`), we will have multiple input sources at some point